content/blog/the-complete-beginners-guide-to-cli-apps-in-v.md

```diff
@@ -0,0 +1,837 @@
+shapes to the screen depending on the options provided. It will be able to
+recognise command-line arguments with the help of the
+[`flag`](https://modules.vlang.io/flag.html) module.
+
+Here's a sample output of our application:
+
+```text
+$ ./geo --shape pyramid --symbol ▲ --size 10
+         ▲
+        ▲▲▲
+       ▲▲▲▲▲
+      ▲▲▲▲▲▲▲
+     ▲▲▲▲▲▲▲▲▲
+    ▲▲▲▲▲▲▲▲▲▲▲
+   ▲▲▲▲▲▲▲▲▲▲▲▲▲
+  ▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲
+ ▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲
+▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲▲
+```
+
+Interested? Let's get started...
+
+## Prerequisites
+
+You should have V installed in your system. Ideally, you should also know
+the basic syntax of V. If you need help with these, refer to
+[this guide](/getting-started-with-v/).
+
+Once you have V installed, make sure you have the latest version of V by
+doing `v up`. It should automatically update your V installation. In case
+of any technical difficulties, you can get in touch with us on the
+[V Discord Server](https://discord.gg/vlang).
+
+The source code for this project is available at
+[this repository](https://github.com/hungrybluedev/geo).
+
+## Initialising the Project
+
+Navigate to a directory that you own. For example, `~/Projects` for Linux/MacOS
+users and `C:\Projects` for Windows users. Then type the following command:
+
+```text
+v new
+```
+
+The tool will prompt you for the project name, description, version, and
+license. Fill in the details as you see fit. Here is an example of what we can
+enter.
+
+```text
+$ v new
+Input your project name: geo
+Input your project description: Sample CLI application that prints shapes to the screen.
+Input your project version: (0.0.0) 0.0.1
+Input your project license: (MIT)
+Initialising ...
+Complete!
+```
+
+Let us take a look at the files that were create by `v new`. In the folder, we
+find a `v.mod` file and a `geo.v` file.
+
+The `v.mod` file looks like this:
+
+```text {linenos=table}
+Module {
+	name: 'geo'
+	description: 'Sample CLI application that prints shapes to the screen.'
+	version: '0.0.1'
+	license: 'MIT'
+	dependencies: []
+}
+```
+
+And this is what `geo.v` contains:
+
+```v {linenos=table}
+module main
+
+fn main() {
+	println('Hello World!')
+}
+```
+
+Let us make sure that we can run the project out of the box by typing the
+command `v run .`:
+
+```text
+$ v run .
+Hello World!
+```
+
+## Accepting Command-Line Arguments
+
+Let us start with how we want to accept command-line arguments. We can make
+use of the in-built [`flag`](https://modules.vlang.io/flag.html) and
+[`os`](https://modules.vlang.io/os.html) modules to do this. In order to use
+these modules in our code, we will need to `import` them.
+
+We will start simple and put everything the `main` function, which is the
+entry point for the application. By default, all variables defined in V
+are immutable. To make a variable mutable, we need to use the `mut` keyword.
+We will make use of this when we define a new `FlagParser` struct using the
```

Comment:
```suggestion
We will make use of this, when we define a new `FlagParser` struct using the
```

Reply:
This comma is probably unnecessary as @JalonSolov commented.